language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java

```diff
@@ -702,6 +707,78 @@ public Optional<TypeSymbol> transform(TableConstructorExpressionNode node) {
 //        return super.transform(interpolationNode);
 //    }
 
+    @Override
+    public Optional<TypeSymbol> transform(FromClauseNode fromClauseNode) {
+        if (context.currentSemanticModel().isEmpty()) {
+            return Optional.empty();
+        }
+
+        // from typed-binding-pattern in <cursor>
+        if (context.getCursorPositionInTree() > fromClauseNode.inKeyword().textRange().endOffset()) {
+            Optional<TypeSymbol> typeSymbol = context.currentSemanticModel().get().typeOf(fromClauseNode.expression());
+
+            if (typeSymbol.isPresent()) {
+                return typeSymbol;
+            }
+            Optional<Symbol> optionalSymbol = context.currentSemanticModel().get()
+                    .symbol(fromClauseNode.typedBindingPattern().bindingPattern());
+            if (optionalSymbol.isEmpty()) {
+                return Optional.empty();
+            }
+            typeSymbol = SymbolUtil.getTypeDescriptor(optionalSymbol.get());
+
+            if (typeSymbol.isEmpty() || typeSymbol.get().typeKind() == TypeDescKind.COMPILATION_ERROR) {
+                return Optional.empty();
+            }
+            Types types = context.currentSemanticModel().get().types();
+            TypeBuilder builder = types.builder();
+            List<TypeSymbol> unionTypeMembers = new ArrayList<>(
+                    List.of(builder.ARRAY_TYPE.withType(typeSymbol.get()).build(),
+                            builder.MAP_TYPE.withTypeParam(typeSymbol.get()).build(),
+                            builder.STREAM_TYPE.withValueType(typeSymbol.get()).build()));
+            if (CommonUtil.getRawType(typeSymbol.get()).typeKind() == TypeDescKind.RECORD) {
+                unionTypeMembers
+                        .add(builder.TABLE_TYPE.withRowType(CommonUtil.getRawType(typeSymbol.get())).build());
+            }
+            if (typeSymbol.get() instanceof StringTypeSymbol) unionTypeMembers.add(types.STRING);
```

Comment:
There are check style errors for this line and the one below. We should use braces with `if` conditions.